Robot Code/src/org/usfirst/frc/team3316/robot/subsystems/Climbing.java

```diff
@@ -0,0 +1,77 @@
+package org.usfirst.frc.team3316.robot.subsystems;
+
+import org.usfirst.frc.team3316.robot.Robot;
+import org.usfirst.frc.team3316.robot.robotIO.DBugSpeedController;
+
+import edu.wpi.first.wpilibj.AnalogPotentiometer;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+
+public class Climbing extends DBugSubsystemCC
+{
+
+	// Sensors
+	private AnalogPotentiometer climbingPot;
+	private DigitalInput climbingSwitch;
+
+	// Actuators
+	private DBugSpeedController climbingMotor1, climbingMotor2, climbingMotor3,
+			climbingMotor4;
+	
+	private DoubleSolenoid climbingSolenoid;
+
+	public Climbing()
+	{
+		// Sensors
+		climbingPot = Robot.sensors.climbingPot;
+
+		climbingSwitch = Robot.sensors.climbingSwitch;
+
+		// Actuators
+		climbingSolenoid = Robot.actuators.climbingSolenoid;
+
+		climbingMotor1 = Robot.actuators.climbingMotor1;
+		climbingMotor2 = Robot.actuators.climbingMotor2;
+		climbingMotor3 = Robot.actuators.climbingMotor3;
+		climbingMotor4 = Robot.actuators.climbingMotor4;
+
+		addSpeedController(climbingMotor1);
+		addSpeedController(climbingMotor2);
+		addSpeedController(climbingMotor3);
+		addSpeedController(climbingMotor4);
+	}
+
+	public void initDefaultCommand()
+	{
+	}
+
+	public boolean setMotors(double v)
+	{
+		return super.setMotors(v);
```

Comment:
Note that this system has slightly different limits. The mechanical limit is that the cable cannot be shorter than the distance between the winch and the rung when the robot assents. And we need to make sure we have the correct direction of movement since theoretically the winch should work in either direction of turning.